### ISSUE TITLE
regex not working with linux file system for ls command with sudo 

### ISSUE THREAD

**Issue by kanhaiyabaranwal** [microsoft/WSL#7867]
ls command works well when the directory has file permission. But, when it doesn't have file permission, and the command is run with sudo, regex doesn't work. I installed mssql server and ran following command.

XXX@Host07:/var/opt$ sudo ls -ltr mssql/data/*.ldf
ls: cannot access 'mssql/data/*.ldf': No such file or directory

However, files exist

-rw-rw---- 1 mssql mssql   8388608 Dec 24 00:57 templog.ldf
-rw-rw---- 1 mssql mssql   8388608 Dec 24 00:57 tempdb.mdf
-rw-rw---- 1 mssql mssql   2097152 Dec 24 01:07 mastlog.ldf

**Comment by danielcranford** [comment]
This is not wsl specific behavior. You will find any Posix file system/shell behaves similarly.

Your problem is the glob pattern (not a regex - they have different syntax) is expanded by the shell BEFORE being passed to the command. If the user running the shell doesn't have access to the directory holding the files matching the glob, the shell CANNOT expand the glob, and leaves it unchanged. The `sudo` elevation only occurs after the shell has already attempted and failed to expand the glob pattern.

TLDR: try running `sudo bash -c "ls -ltr mssql/data/*.ldf"` to create a privileged shell process which can expand the glob before passing it to `ls`

**Comment by kanhaiyabaranwal** [comment]
> This is not wsl specific behavior. You will find any Posix file system/shell behaves similarly.
> 
> Your problem is the glob pattern (not a regex - they have different syntax) is expanded by the shell BEFORE being passed to the command. If the user running the shell doesn't have access to the directory holding the files matching the glob, the shell CANNOT expand the glob, and leaves it unchanged. The `sudo` elevation only occurs after the shell has already attempted and failed to expand the glob pattern.
> 
> TLDR: try running `sudo bash -c "ls -ltr mssql/data/*.ldf"` to create a privileged shell process which can expand the glob before passing it to `ls`

**Comment by kanhaiyabaranwal** [comment]
thanks it helped
